### ISSUE TITLE
Cannot load darwin/arm64/go(1.22,1.23)/cgo binary

### ISSUE THREAD

**Issue by Zxilly** [goretk/gore#96]
Here's example:

https://github.com/Zxilly/go-testdata/releases/download/latest/bin-darwin-1.22-arm64-strip-cgo
https://github.com/Zxilly/go-testdata/releases/download/latest/bin-darwin-1.22-arm64-strip-pie-cgo
https://github.com/Zxilly/go-testdata/releases/download/latest/bin-darwin-1.23-arm64-strip-cgo
https://github.com/Zxilly/go-testdata/releases/download/latest/bin-darwin-1.23-arm64-strip-pie-cgo

The search of `*pcHeader` magic will return -1 in `findRuntimeText`

**Comment by Zxilly** [comment]
![image](https://github.com/user-attachments/assets/943c9ccd-7f95-4d5f-8658-a5bd4248411a)
![image](https://github.com/user-attachments/assets/abd3fed1-d763-40ad-800f-f708794350f6)

Memory addresses on macOS seem to have a strange byte order and only happen when cgo is enabled

**Comment by Zxilly** [comment]
![image](https://github.com/user-attachments/assets/bfbc63fd-7c12-44e0-b83c-be5caa05c043)

This behavior can be reproduce

**Comment by Zxilly** [comment]
I test multi go version, seems this is always a error

```
// format
// pclntab addr value
// generate search hash
// real value in the binary

19
pclntabAddr 4295817152
c0 f7 0c 00 01 00 00 00 
c0 f7 0c 00 00 00 10 00

20
pclntabAddr 4295805984
20 cc 0c 00 01 00 00 00 
20 cc 0c 00 00 00 10 00

21
pclntabAddr 4298061728
a0 37 2f 00 01 00 00 00 
a0 37 2f 00 00 00 10 00

22
pclntabAddr 4297689728
80 8a 29 00 01 00 00 00 
80 8a 29 00 00 00 10 00

23
pclntabAddr 4297729600
40 26 2a 00 01 00 00 00 
40 26 2a 00 00 00 10 00
```

The problem was because prior to go 1.21, the runtime.text symbol would still be present in the binary even if the -s flag was passed, masking the problem. This bug was fixed in 1.22, causing this error to be exposed.
